compositor_render/src/scene/rescaler_component/layout.rs

```diff
@@ -0,0 +1,111 @@
+use std::time::Duration;
+
+use compositor_common::util::align::{HorizontalAlign, VerticalAlign};
+
+use crate::{
+    scene::{layout::StatefulLayoutComponent, ResizeMode, Size, StatefulComponent},
+    transformations::layout::{LayoutContent, NestedLayout},
+};
+
+use super::RescalerComponentParam;
+
+impl RescalerComponentParam {
+    pub(super) fn layout(
+        &self,
+        size: Size,
+        child: &StatefulComponent,
+        pts: Duration,
+    ) -> NestedLayout {
+        let child_width = child.width(pts);
+        let child_height = child.height(pts);
+        match (child_width, child_height) {
+            (None, None) => self.layout_with_scale(size, child, pts, 1.0),
+            (None, Some(child_height)) => {
+                self.layout_with_scale(size, child, pts, size.height / child_height)
+            }
+            (Some(child_width), None) => {
+                self.layout_with_scale(size, child, pts, size.width / child_width)
+            }
+            (Some(child_width), Some(child_height)) => {
+                let scale = match self.mode {
+                    ResizeMode::Fit => {
+                        f32::min(size.width / child_width, size.height / child_height)
+                    }
+                    ResizeMode::Fill => {
+                        f32::max(size.width / child_width, size.height / child_height)
+                    }
+                };
+                self.layout_with_scale(size, child, pts, scale)
+            }
+        }
+    }
+
+    fn layout_with_scale(
+        &self,
+        size: Size,
+        child: &StatefulComponent,
+        pts: Duration,
+        scale: f32,
+    ) -> NestedLayout {
+        let (content, children, child_nodes_count) = match child {
+            StatefulComponent::Layout(layout_component) => {
+                let children_layouts = layout_component.layout(
+                    Size {
+                        width: size.width / scale,
+                        height: size.height / scale,
```

Comment:
[NIT]
I think it's more intuitive to multiply by scale param, than divide by it. You would also need to change the division order in lines 32 and 35 in such case.

Reply:
yes that is how scale works here, but in this specific case I'm doing transformation in other direction. I have a size after scalling and want to calculate what size it should have before scalling